src/Domain.LinnApps/Parts/PartService.cs

```diff
@@ -215,6 +220,27 @@ public Part CreateFromSource(int sourceId, int createdBy)
             return this.partRepository.FindBy(p => p.PartNumber == source.PartNumber);
         }
 
+        public IEnumerable<Part> GetDeptStockPalletParts()
+        {
+            var parts = this.partRepository.FindAll();
+            var stockLocators = this.stockLocatorRepository.FindAll();
+            return from p in parts
+                    where (from s in stockLocators 
+                           where (s.PartNumber == p.PartNumber && s.StockPoolCode.Equals("LINN DEPT"))
+                           select s.PartNumber)
+                        .Contains(p.PartNumber)
+                        ||
+                        ((p.StockControlled.Equals("N") 
+                          && (p.BaseUnitPrice == 0 || p.BaseUnitPrice == null)
+                          && (p.LinnProduced.Equals("N") || p.LinnProduced == null))
+                        &&
+                        !(from s in stockLocators
+                         where s.PartNumber == p.PartNumber
+                         select s.PartNumber)
+                        .Contains(p.PartNumber))
+                    select p;
```

Comment:
Could this be split up to make it a bit clearer? Just thinking it might be hard to follow / debug